testlib/src/lib.rs

```diff
@@ -44,3 +47,31 @@ pub fn annotation(
         Err(err) => err.to_compile_error().into(),
     }
 }
+
+#[proc_macro_attribute]
+pub fn outer(
+    _attr: proc_macro::TokenStream,
+    item: proc_macro::TokenStream,
+) -> proc_macro::TokenStream {
+    let item = parse_macro_input!(item as syn::ItemFn);
+    let annotation_attr =
+        item.attrs.iter().find(|attr| attr.path().is_ident("annotation"));
+    let annotation_attr = annotation_attr.expect("annotation attribute found");
+    let l = match &annotation_attr.meta {
+        syn::Meta::List(l) => l,
+        _ => {
+            panic!("annotation attribute must be a list")
+        }
+    };
```

Comment:
```suggestion
    let syn::Meta::List(l) = &annotation_attr.meta else {
        panic!("annotation attribute must be a list");
    };
```

Would this work?

Also, one of the very early pieces of coding advice I recall is "don't use `l` as a variable name" because it's too easily confused with e.g. a 1 or something else. Do what you will with that!